platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/ConfigurableFileTreeCodec.kt

```diff
@@ -30,14 +32,17 @@ class ConfigurableFileTreeCodec(
     private val fileCollectionFactory: FileCollectionFactory
 ) : Codec<ConfigurableFileTree> {
     override suspend fun WriteContext.encode(value: ConfigurableFileTree) {
+        value as DefaultConfigurableFileTree
```

Comment:
💅 `require(value is DefaultConfigurationFileTree)` is more expressive and the failure mode is clearer (a precondition violation instead of a class cast exception)